todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/TasksLocalDataSource.kt

```diff
@@ -94,9 +94,11 @@ class TasksLocalDataSource private constructor(context: Context) : TasksDataSour
             close()
         }
         db.close()
-        if(task!=null){
```

Comment:
I had a change of heart about this... I prefer reverting to what you originally had, but use `also` instead of `let` so the left side of `?:` evaluates to the receiver.
Like this: 
task?.also { callbackOnTaskLoaded(it) } ?: callback.onDataNotAvailable()

That way, onDataNotAvailable is always called if `task` is null. The current way technically could silently fail if `task` becomes null between `if (task != null)` and `task?.let ...` (though impossible in this sample).

Reply:
Done!!
Yes you are right, current code will definitely work for this sample but your suggestion is better as it makes consideration for multi threaded scenarios.